### PR TITLE
Added Redirect Capability

### PR DESCRIPTION
**Pull request by tr-cimpress**
Added method override to provideAuthService to allow for specifying a
redirect and a homepath URL.
Added constructor overrides to allow for passthrough of the redirect and
homepath URLs a well as the ability to inject the router for the
application.

### CODE REVIEW

File: auth0-lock.ts [hunk 1]
```diff
@@ -9,13 +10,20 @@ declare var Auth0Lock: any;
 export class AuthService {
   // Configure Auth0
   lock = new Auth0Lock(this.clientId, this.domain, {
-    closable: false // Users are required to log in
+    closable: false, // Users are required to log in
+    auth: {
+      redirectUrl: (this.redirectPath == null ? location.origin : location.origin + this.redirectPath),
```
Comment (slowvincy) [inline auth0-lock.ts:15]: What happens if you just say `location.origin + this.redirectPath` here?

File: auth0-lock.ts [hunk 2]
```diff
@@ -9,13 +10,20 @@ declare var Auth0Lock: any;
...
+    closable: false, // Users are required to log in
+    auth: {
+      redirectUrl: (this.redirectPath == null ? location.origin : location.origin + this.redirectPath),
+      responseType: 'token'
+    }
   });
 
-  constructor(private clientId: string, private domain: string) {
+  constructor( clientId: string,  domain: string);
+  constructor( clientId: string,  domain: string, homePath: string, redirectPath: string, router: Router);
```
Comment (slowvincy) [inline auth0-lock.ts:21]: I wonder why is this signature necessary?

File: auth0-lock.ts [hunk 3]
```diff
@@ -53,9 +61,22 @@ export class AuthService {
...
 
-export function provideAuthService(clientId: string, domain: string): Provider {
-  return {
-    provide: AuthService,
-    useValue: new AuthService(clientId, domain)
-  };
+export function provideAuthService(clientId: string, domain: string): Provider;
+export function provideAuthService(clientId: string, domain: string, homePath: string, redirectPath: string): Provider;
+export function provideAuthService(clientId: string, domain: string, homePath?: string, redirectPath?: string): Provider {
+  if(redirectPath == null){
```
Comment (slowvincy) [inline auth0-lock.ts]: Should this check be on `homePath`? That is the one that requires `router` dependency.

File: auth0-lock.ts (hunk 1 above)
Comment (tr-cimpress) [reply auth0-lock.ts:15]: That would probably work as well, string concatenation in Typescript should do the same thing either way.

File: auth0-lock.ts (hunk 3 above)
Comment (tr-cimpress) [reply auth0-lock.ts]: It should be, I'll change that.

File: auth0-lock.ts (hunk 2 above)
Comment (tr-cimpress) [reply auth0-lock.ts:21]: This seems to be the recommended method of overloading methods in Typescript.

File: auth0-lock.ts (hunk 1 above)
Comment (tr-cimpress) [reply auth0-lock.ts:15]: Turns out this doesn't work like that. For example if `location.origin` is http:\\example.com and `this.redirectPath` isn't set, the results of  `location.origin + this.redirectPath` will end up as __http:\\example.comundefinded__

**Review by slowvincy — APPROVED** [review]

**Pull request merged**